### PR TITLE
Dev Docker with pip

### PR DESCRIPTION
**Pull request by GernotMaier**
Two changes:

- use pip instead conda in container
- update  the readme adding the executation of `entrypoint.sh` to the testing step.

Closely related to #42 (tried to sync as much as possible)

### CODE REVIEW

**Comment by GernotMaier** [comment]
@orelgueta  @VictorBarbosaMartins 

This is still in draft, as test are not working (and I am not sure what is going wrong):

```
docker run --rm -it -v "$(pwd)/external:/workdir/external" simtools-dev bash -c "$(cat ./entrypoint.sh) && bash"

 pytest --no-cov
ImportError while loading conftest '/workdir/external/simtools/tests/conftest.py'.
tests/conftest.py:12: in <module>
    from simtools.corsika.corsika_output import CorsikaOutput
simtools/corsika/corsika_output.py:11: in <module>
    from corsikaio.subblocks import event_header, get_units_from_fields, run_header
E   ImportError: cannot import name 'get_units_from_fields' from 'corsikaio.subblocks' (/workdir/conda/lib/python3.9/site-packages/corsikaio/subblocks/__init__.py)
```

Suggestions to solve this are welcome!

**Comment by VictorBarbosaMartins** [comment]
Hi @GernotMaier .

This is exactly the error I was getting while developing the CorsikaOutput module until I realized that `pyeventio` was actually installing an old version of `corsikaio`, which did not have the `get_units_from_fields ` function.

Since Max corrected the requirements in setup.py in `eventio` and corrected a unit that was wrong in `corsikaio`, we need now at least `corsikaio v 0.3.3` and `eventio 1.11.0`. (maybe we should add this version requirement in simtools)

I've checked the dockerfile and its seems it points still to a "master" branch and not to "main" (not sure though whether these issues are related to each other):
https://github.com/gammasim/containers/blob/cbd69ce00f620e17a3a5426ce73fe892cebe1d75/dev/Dockerfile#L35C18-L35C18

**Comment by GernotMaier** [comment]
This is now weird: on mac os and on the wgs, we get eventio 1.11.0 with the correct environment file from the main branch and everything is fine. In the dev Docker, it is eventio 1.7.0.

File: dev/Dockerfile [hunk 1]
```diff
@@ -13,51 +13,45 @@ RUN apt-get update && apt-get install -y \
...
 
 RUN apt-get update && apt-get install -y \
+    bc \
     bzip2 \
     gfortran \
     libgsl-dev \
-    bc \
-    unzip && \
+    python3-pip \
+    unzip \
```
Comment (orelgueta) [inline dev/Dockerfile]: Are you sure we still need unzip here?

**Review by orelgueta — DISMISSED** [review]
Looks all good.

**Review by VictorBarbosaMartins — CHANGES_REQUESTED** [review]
Hi @GernotMaier.

It looks very good, but I would like to include the `pip install -e .` at the end that allows us to work on the code afterwards (development). Every time I open a new container I have to type it and sometimes I forget the `-e` which then leads to very strange behaviour.  Thanks!

**Comment by GernotMaier** [comment]
@VictorBarbosaMartins - call `pip install -e .` in https://github.com/gammasim/containers/blob/usage_update/dev/entrypoint.sh (and it needs to be called in your simtools directory, which we don't have in the container)

**Review by VictorBarbosaMartins — APPROVED** [review]

File: dev/Dockerfile (hunk 1 above)
Comment (GernotMaier) [reply dev/Dockerfile]: No, we don't need it. I've removed it now.

**Pull request merged**